portfolio/src/main/webapp/script.js

```diff
@@ -293,9 +293,33 @@ function deleteSpecificComment(comment) {
 async function addComment() {
   const params = new URLSearchParams();
   const text = document.getElementById('text').value;
+  if (text === '') {
+    document.getElementById('text').value = '';
+    alert('Your message is empty. Please try again');
+    return;
+  }
   params.append('text', text);	
-  await fetch('/add-comment', {method: 'POST', body: params});
-  getCommentSection();
+  fetch(
+  'https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=API_KEY',
+    {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({comment: {text: text}, languages: [], requestedAttributes: { TOXICITY: {} }})
+    }
+    ).then(async response => response.json()).then(async data => {
+      const score = data.attributeScores.TOXICITY.summaryScore.value;
+      if (score > 0.8) {
+        document.getElementById('text').value = '';
+        alert('Your message is too toxic. Please try again');
+        return;
+      }
+      else {
+        params.append('toxicity', score);
+        await fetch('/add-comment', {method: 'POST', headers: {'Content-Type': 'application/json'}, body: JSON.stringify({text: text, toxicity: score})});
+        getCommentSection();
+      }
+    }
+  );
```

Comment:
This isn't really part of the week 4 work, but as you do more with APIs you might want to add `catch` functions to handle any errors (for example if Perspective is down, or if the user submits a comment in an unsupported language).  This will let you fail gracefully (maybe with an error message) if something unexpected happens.  Here is a bit of documentation on how to handle errors in JavaScript: https://javascript.info/promise-error-handling

Reply:
Okay, I will use this next time!